### DISCUSSION TITLE
Best practice

### DISCUSSION THREAD

**Discussion by xu2xulim** [deta/cloud-docs#317] [Help]
What is the best practice to use Deta Base to log completion report (success or failure) such that it appears **chronologically** on the UI. I am currently using the system generated key but I do have a date time stamp as part of the log. An expiry is set to delete entry in X days. Presently, it is appearing randomly and I have to check my own timestamp. Thanks.

**Comment by abdelhai** [comment]
you could use a unix timestamp (or any datetime string for that matter) as the `key`. this will make sure your items are automatically sorted.

**Reply by xu2xulim** [reply]
Any particular reason why this is not done by Deta when we do not supply a key during an insert or put?

**Reply by abdelhai** [reply]
deta can't guarantee ordered items in general, especially at scale, so we don't bother (deta base is eventually consistent). it's up to the developer when they use base to decide if they want to trust the consistency model or not. at a small scale, it doesn't really matter as you might notice inconsistencies.

in short, the solution i suggested doesn't really work at a large scale but is fine for small apps. for large scale apps (>100s ops/seq), base is not the right choice if you want ordered data, that's what relational/ACID-compliant databases excel at.

**Reply by xu2xulim** [reply]
Thank you for the explanation. Using query and ?pfx helps seems to help with what I am hoping to achieve.
